src/Writer/Common.php

```diff
@@ -22,7 +22,7 @@ class Common extends Writer implements WriterInterface
     /** @var \PDO */
     protected $db;
 
-    public function createConnection(array $dbParams): \PDO
+    public function createConnection(array $dbParams)
```

Comment:
I think you can return mixed types by using block comments: https://github.com/keboola/db-extractor-common/blob/master/src/Keboola/DbExtractor/Extractor/Extractor.php#L93